### PR TITLE
update scmType to VSTSRM for web app

### PR DESCRIPTION
**Pull request by dikhakha**

### CODE REVIEW

File: Tasks/Common/azurerest-common/azurerestutility.ts [hunk 1]
```diff
@@ -218,6 +218,36 @@ export async function getAzureRMWebAppConfigDetails(SPN, webAppName: string, res
...
+             '/providers/Microsoft.Web/sites/' + webAppName + slotUrl +  '/config/web?' + azureApiVersion;
+	
+    tl.debug('Updating config details at: ' + configUrl);
+	
+    httpObj.send('PATCH', configUrl, configDetails, headers, (error, response, body) =>{
+		if(error){
+			deferred.reject(error);
+		}
+		else if(response.statusCode === 200) {
+			deferred.resolve(tl.loc("SuccessfullyUpdatedAzureRMWebAppConfigDetails"));
```
Comment (sachinma) [inline Tasks/Common/azurerest-common/azurerestutility.ts]: Dont return the message from the method. Do this check in the caller.

File: Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts [hunk 2]
```diff
@@ -194,4 +195,22 @@ async function DeployUsingKuduDeploy(webDeployPkg, azureWebAppDetails, publishin
     }
 }
 
+async function updateConfigDetails(SPN, webAppName: string, resourceGroupName: string, deployToSlotFlag: boolean, slotName: string) {
+    try {
+        var configDetails = await azureRESTUtility.getAzureRMWebAppConfigDetails(SPN, webAppName, resourceGroupName, deployToSlotFlag, slotName);
+        if(configDetails.properties.scmType === "None") {
```
Comment (sachinma) [inline Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts]: should it be case insensitive.

Comment (dikhakha) [inline Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts]: done

File: Tasks/Common/azurerest-common/azurerestutility.ts (hunk 1 above)
Comment (dikhakha) [inline Tasks/Common/azurerest-common/azurerestutility.ts]: done

File: Tasks/AzureRmWebAppDeployment/task.json [hunk 3]
```diff
@@ -285,6 +285,8 @@
         "WebAppDoesntExist": "Webapp '%s' doesn't exist. Webapp should exist before deployment.",
         "EncodeNotSupported": "Detected file encoding of the file %s as %s. Variable substitution and Transformation is not supported with file encoding %s. Supported encodings are UTF-8 and UTF-16 LE.",
         "UnknownFileEncodeError": "Unable to detect encoding of the file %s (typeCode: %s). Supported encodings are UTF-8 and UTF-16 LE.",
-        "ShortFileBufferError": "File buffer is too short to detect encoding type : %s"
+        "ShortFileBufferError": "File buffer is too short to detect encoding type : %s",
+        "UnabletoupdateAzureRMWebAppConfigDetails": "Error Occurred : [Status Code : '%s', Error Message: '%s']",
```
Comment (sachinma) [inline Tasks/AzureRmWebAppDeployment/task.json]: Make it camel case

File: Tasks/Common/azurerest-common/azurerestutility.ts [hunk 4]
```diff
@@ -218,6 +218,36 @@ export async function getAzureRMWebAppConfigDetails(SPN, webAppName: string, res
...
+	
+    httpObj.send('PATCH', configUrl, configDetails, headers, (error, response, body) =>{
+		if(error){
+			deferred.reject(error);
+		}
+		else if(response.statusCode === 200) {
+			deferred.resolve();
+		}
+		else {
+			deferred.reject(tl.loc('UnabletoupdateAzureRMWebAppConfigDetails', response.statusCode, response.statusMessage));
```
Comment (sachinma) [inline Tasks/Common/azurerest-common/azurerestutility.ts]: Try not to use the localization variables in here

File: Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts [hunk 5]
```diff
@@ -194,4 +195,24 @@ async function DeployUsingKuduDeploy(webDeployPkg, azureWebAppDetails, publishin
     }
 }
 
+async function updateConfigDetails(SPN, webAppName: string, resourceGroupName: string, deployToSlotFlag: boolean, slotName: string) {
```
Comment (dikhakha) [inline Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts]: yes will do that.

File: Tasks/AzureRmWebAppDeployment/task.json (hunk 3 above)
Comment (dikhakha) [inline Tasks/AzureRmWebAppDeployment/task.json]: done. Changed the name to "FailedToUpdateAzureRMWebAppConfigDetails"

File: Tasks/Common/azurerest-common/azurerestutility.ts (hunk 4 above)
Comment (dikhakha) [inline Tasks/Common/azurerest-common/azurerestutility.ts]: done

**Pull request merged**